presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoMetadata.java

```diff
@@ -244,6 +246,20 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 localProperties.build());
     }
 
+    @Override
+    public Optional<LimitApplicationResult<ConnectorTableHandle>> applyLimit(ConnectorSession session,
+	    ConnectorTableHandle table, long limit) {
+	MongoTableHandle handle = (MongoTableHandle) table;
+
+	if (handle.getLimit().isPresent() && handle.getLimit().getAsLong() <= limit) {
+	    return Optional.empty();
+	}
+
+	handle = new MongoTableHandle(handle.getSchemaTableName(), handle.getConstraint(), OptionalLong.of(limit));
+
+	return Optional.of(new LimitApplicationResult<>(handle, true));
```

Comment:
Can we inline the `handle` variable 
```suggestion
	return Optional.of(new LimitApplicationResult<>(
	        new MongoTableHandle(handle.getSchemaTableName(), handle.getConstraint(), OptionalLong.of(limit)),
	        true));
```